### PR TITLE
Feature/ui widgets

### PR DESCRIPTION
**Pull request by ooms97**
This PR includes many of the widgets on the app, such as:
 - The search menu and building list (not functional).
 - The side drawer.
 - The campus toggle button (not functional).
 - The current location button.
 - The location indicator.
 - The navigation button and page (not functional).

### CODE REVIEW

File: lib/views/go_to_page.dart [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        border: OutlineInputBorder(borderRadius: BorderRadius.circular(32.0))),
+  );
+}
+
+class _GoToPageState extends State<GoToPage> {
+  //To be able to save the selected suggestion into the text field
+  final TextEditingController _originTypeAheadController =
+      TextEditingController();
+  final TextEditingController _destinTypeAheadController =
+      TextEditingController();
```
Comment (MrJCipherButtles) [inline lib/views/go_to_page.dart]: Can you edit this to be more readable?

File: lib/views/go_to_page.dart [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                          contentPadding:
+                              EdgeInsets.fromLTRB(15.0, 12.0, 10.0, 12.0),
+                          hintText: "destination...",
+                          border: OutlineInputBorder(
+                              borderRadius: BorderRadius.circular(10.0)),
+                        ),
+                        controller: this._destinTypeAheadController,
+                      ),
+                      suggestionsCallback: (pattern) async {
+                        var roomTitles = new List<String>();
```
Comment (MrJCipherButtles) [inline lib/views/go_to_page.dart]: Most likely going to want to make this a List of RoomCoordinate. You should also add a TODO comment similar to line 78

File: lib/views/go_to_page.dart [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+  @override
+  Widget build(BuildContext context) {
+    List<String> originDestination = [];
+    FocusNode destinationTextField =
+    new FocusNode(); // this focus node will be used for the second text field (destination)
+    return Container(
+      child: Material(
+        child: Container(
+          margin: new EdgeInsets.only(left: 15.0, top: MediaQuery.of(context).padding.top + 5.0, right: 15.0),
+          //color: Color.fromRGBO(147, 0, 44, 1),
```
Comment (MrJCipherButtles) [inline lib/views/go_to_page.dart]: Why is this commented out? Should it be deleted?

File: lib/views/go_to_page.dart [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+                      },
+                      itemBuilder: (context, suggestion) {
+                        return Card(
+                          elevation: 2,
+                          child: ListTile(
+                            leading: Icon(Icons.place,
+                                color: Color.fromRGBO(147, 0, 47, 1)),
+                            title: Text(suggestion),
+
+                            //subtitle: Text('\$${suggestion['price']}'),
```
Comment (MrJCipherButtles) [inline lib/views/go_to_page.dart]: Can this comment be deleted? Or is it there for a future reminder?

File: lib/views/go_to_page.dart [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+                        var roomTitles = new List<String>();
+                        for (var room in []) {
+                          if (room
+                              .getTitle()
+                              .toUpperCase()
+                              .startsWith(pattern.toString().toUpperCase())) {
+                            roomTitles.add(room.getTitle());
+                          }
+                        }
+                        return roomTitles; //await BackendService.getSuggestions(pattern);
```
Comment (MrJCipherButtles) [inline lib/views/go_to_page.dart]: Is the comment there as a reminder?

File: lib/views/go_to_page.dart [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+                      },
+                      itemBuilder: (context, suggestion) {
+                        return Card(
+                          elevation: 2,
+                          child: ListTile(
+                            leading: Icon(Icons.place,
+                                color: Color.fromRGBO(147, 0, 47, 1)),
+
+                            title: Text(suggestion),
+                            //subtitle: Text('\$${suggestion['price']}'),
```
Comment (MrJCipherButtles) [inline lib/views/go_to_page.dart]: Should this comment be deleted?

File: lib/views/home_page.dart [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+  Widget build(BuildContext context) {
+
+  if (_cameraPosition == null) {
+      _cameraPosition = CameraPosition(target: LatLng(0, 0));
+  }
+
+    return Scaffold(
+        body: Stack(
+              children: <Widget>[
+                // Replace this container with your Map widget
```
Comment (MrJCipherButtles) [inline lib/views/home_page.dart]: What is your Map widget? Is this meant for a reminder?

File: lib/widgets/drawer.dart [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+import 'package:flutter/material.dart';
+
+class SidebarDrawer extends StatefulWidget {
+  @override
+  _SidebarDrawerState createState() => _SidebarDrawerState();
+}
+
+class _SidebarDrawerState extends State<SidebarDrawer> {
+  Widget build(BuildContext context) {
+    //DisabilityNotifier disabilityNotifier = Provider.of<DisabilityNotifier>(context);
```
Comment (MrJCipherButtles) [inline lib/widgets/drawer.dart]: Why is this commented out? Should it be there?

File: lib/widgets/search/building_manager.dart [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+                    Navigator.pop(context);
+                  },
+                );
+              },
+              separatorBuilder: (context, index) {
+                return Divider();
+              },
+            )));
+  }
+}
```
Comment (MrJCipherButtles) [inline lib/widgets/search/building_manager.dart]: Can you format this better?

File: lib/widgets/search/main_search_bar.dart [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+                color: Color.fromRGBO(147, 35, 57, 1),
+                shape: RoundedRectangleBorder(
+                    borderRadius: new BorderRadius.circular(50)),
+                onPressed: () {
+                  setState(() {
+                    (campus == 'SGW') ? (campus = 'LOY') : (campus = 'SGW');
+                  });
+                },
+              ),
+              )],
```
Comment (MrJCipherButtles) [inline lib/widgets/search/main_search_bar.dart]: This looks like it is one indent too much. Is it?

File: lib/widgets/search/main_search_bar.dart [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+        ),
+      ),
+    ]);
+  }
+}
+
+class PositionedFloatingSearchBar extends SearchDelegate<String> {
+
+final Function(LatLng) latlng;
+PositionedFloatingSearchBar({this.latlng});
```
Comment (MrJCipherButtles) [inline lib/widgets/search/main_search_bar.dart]: Missing indentation?

**Comment by ChesterLY** [comment]
Looks good to me

File: lib/widgets/search/main_search_bar.dart [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+          close(context, null);
+        });
+  }
+
+  @override
+  Widget buildSuggestions(BuildContext context) {
+    final suggestionList = query.isNotEmpty
+        ? []
+        : []
+            .where((p) => p.getRoomNumber().startsWith(query.toUpperCase()))
```
Comment (MrJCipherButtles) [inline lib/widgets/search/main_search_bar.dart]: This looks off. Getter for RoomCoordinate room number is roomId

File: lib/widgets/search/main_search_bar.dart [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+            .toList();
+    return query.isEmpty ? SearchMenuListOption() : ListView.builder(
+      itemBuilder: (context, index) => ListTile(
+        onTap: () {
+        },
+        leading: Icon(Icons.place),
+        title: Text(suggestionList[index].getRoomNumber()),
```
Comment (MrJCipherButtles) [inline lib/widgets/search/main_search_bar.dart]: This looks off. Getter for RoomCoordinate room number is roomId

File: lib/widgets/search/main_search_bar.dart [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+      ),
+      itemCount: suggestionList.length,
+    );
+  }
+
+  @override
+  Widget buildResults(BuildContext context) {
+    // TODO: implement buildResults
+    return null;
+  }
```
Comment (MrJCipherButtles) [inline lib/widgets/search/main_search_bar.dart]: Does this widget have a purpose?

File: lib/widgets/search/search_menu_suggestions.dart [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+                  title: Text(_myRoomList[index].roomId),
+                  onTap: () {},
+                );
+              },
+              separatorBuilder: (context, index) {
+                return Divider();
+              },
+              )
+            ));
+  }
```
Comment (MrJCipherButtles) [inline lib/widgets/search/search_menu_suggestions.dart]: Indentation looks off

File: lib/widgets/search/search_menu.dart [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      child: Column(
+        children: <Widget>[
+          Row(children: <Widget>[
+      ]),
+        ]));
+  }
```
Comment (MrJCipherButtles) [inline lib/widgets/search/search_menu.dart]: Indentation looks off

File: lib/views/go_to_page.dart (hunk 1 above)
Comment (ooms97) [reply lib/views/go_to_page.dart]: Changed these two lines to be more readable.

File: lib/views/go_to_page.dart (hunk 3 above)
Comment (ooms97) [reply lib/views/go_to_page.dart]: Removed this line.

File: lib/views/go_to_page.dart (hunk 4 above)
Comment (ooms97) [reply lib/views/go_to_page.dart]: It was deleted.

**Comment by ooms97** [comment]
@MrJCipherButtles I added all the changes as requested and auto-formatted the entire lib directory using flutter format.

**Comment by MrJCipherButtles** [comment]
> @MrJCipherButtles I added all the changes as requested and auto-formatted the entire lib directory using flutter format.

@ooms97 Great I didn't know that option existed! All should be well-formatted now 👍

File: lib/views/go_to_page.dart (hunk 1 above)
Comment (ooms97) [reply lib/views/go_to_page.dart]: The flutter format defaults to this.

File: lib/views/go_to_page.dart (hunk 5 above)
Comment (ooms97) [reply lib/views/go_to_page.dart]: Removed the comment.

File: lib/widgets/search/main_search_bar.dart (hunk 12 above)
Comment (ooms97) [reply lib/widgets/search/main_search_bar.dart]: Modified this to match current models.

File: lib/widgets/search/main_search_bar.dart (hunk 13 above)
Comment (ooms97) [reply lib/widgets/search/main_search_bar.dart]: Changed to roomId.

File: lib/widgets/search/main_search_bar.dart (hunk 14 above)
Comment (ooms97) [reply lib/widgets/search/main_search_bar.dart]: The override is necessary. Implementation is required for future PR.

File: lib/widgets/search/search_menu_suggestions.dart (hunk 15 above)
Comment (ooms97) [reply lib/widgets/search/search_menu_suggestions.dart]: Indentation fixed with flutter format.

File: lib/views/go_to_page.dart (hunk 6 above)
Comment (ooms97) [reply lib/views/go_to_page.dart]: Deleted this comment.

File: lib/views/home_page.dart (hunk 7 above)
Comment (ooms97) [reply lib/views/home_page.dart]: Deleted this comment.

File: lib/widgets/drawer.dart (hunk 8 above)
Comment (ooms97) [reply lib/widgets/drawer.dart]: Deleted this comment.

File: lib/widgets/search/main_search_bar.dart (hunk 10 above)
Comment (ooms97) [reply lib/widgets/search/main_search_bar.dart]: Formatting fixed with flutter format.

File: lib/widgets/search/main_search_bar.dart (hunk 11 above)
Comment (ooms97) [reply lib/widgets/search/main_search_bar.dart]: Formatting fixed with flutter format.

**Pull request merged**